app/models/mixins/supports_feature_mixin.rb

```diff
@@ -105,6 +105,32 @@ def supports(feature, &block)
       define_supports_feature_methods(feature, &block)
     end
 
+    # delegate to a child class whether a feature is supported
+    # Author => Author::Post
+    # will only delegate to a class level supports
+    def delegate_supports(feature, target_name, target_feature)
+      method_name = "supports_#{feature}?"
```

Comment:
I think we are trying (in the future) to limit the number of methods each Class/Instance have - so that we only implement somehow a method called `supports?` taking the feature as a parameter.  The way it is currently implemented (in the SupportsFeatureMixin, not just in this PR) creates unneeded method bloat that we are trying to get away from.

Reply:
ok, I was just following the pattern. The goal is to us a hash or something?

Reply:
it is on my TODO list - but I am not against anybody else taking a shot at it
the other issues on my TODO list are
- improving `unsupported_reason_add` caller to not need to pass feature when in a `supports` block
- improving `unsupported_reason_add` to noop if an unsupported reason is already provided for that feature

Reply:
I keep thinking that supports is a hash on the class of feature to reason instead of a method.
the value of
- true means it passes
- false means there is no reason but it fails
- string for the reason it fails
- lambda for "it's complicated"